### PR TITLE
Added SQL Support & Date/Time Utils

### PR DESCRIPTION
**Pull request by kotlinsyntax**
Added SQL support including POJO's and a QueryBuilder

### CODE REVIEW

**Comment by kotlinsyntax** [comment]
[Add support for multiple WHERE and JOIN clauses](https://github.com/flytegg/twilight/pull/48/commits/5feaa7b10c8981501e03c2a9f1cd8793fe56e9fc)

**Review by joshbker — CHANGES_REQUESTED** [review]
Hey, could you add some basic tests for your new things added, such as SQL and DateTime utils. Then I'll merge into a new branch and do some refactoring to keep consistent. Thanks :)

**Comment by kotlinsyntax** [comment]
> Hey, could you add some basic tests for your new things added, such as SQL and DateTime utils. Then I'll merge into a new branch and do some refactoring to keep consistent. Thanks :)

Alright how do you want me to do that? like a test table and run some queries on it?

**Comment by kotlinsyntax** [comment]
Updated the README for SQL

File: src/main/kotlin/gg/flyte/twilight/data/sql/AsyncSQLWrapper.kt
Comment (joshbker) [inline src/main/kotlin/gg/flyte/twilight/data/sql/AsyncSQLWrapper.kt:1]: Please can you document that this exists and any differences from the SQLWrapper

File: src/main/kotlin/gg/flyte/twilight/data/sql/SQLSerializable.kt
Comment (joshbker) [inline src/main/kotlin/gg/flyte/twilight/data/sql/SQLSerializable.kt:1]: Please could you document that this exists and how to use it for easy table creation, and POJO saving/loading.

File: src/main/kotlin/gg/flyte/twilight/data/sql/SQLSerializable.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            "boolean", "kotlin.Boolean", "java.lang.Boolean" -> "BOOLEAN"
+            "java.util.Date", "java.time.LocalDate", "java.time.LocalTime", "java.time.LocalDateTime", "java.sql.Date", "java.sql.Time", "java.sql.Timestamp" -> "TIMESTAMP"
+            "java.net.InetAddress" -> if(dialect == Dialect.POSTGRES) "INET" else "VARCHAR(43)"
+            "java.util.UUID" -> if(dialect == Dialect.POSTGRES) "UUID" else "VARCHAR(36)"
+            // this needs more testing. "java.util.ArrayList", "java.util.List", "java.util.Vector", "kotlin.collections.List", "kotlin.collections.ArrayList" -> if(dialect == Dialect.POSTGRES) "ARRAY" else "STRING"
+            else -> throw IllegalArgumentException("Error: Unsupported type: $kotlinType")
+        }
+    }
+
+    private fun pluralize(word: String, count: Int = 2): String {
```
Comment (joshbker) [inline src/main/kotlin/gg/flyte/twilight/data/sql/SQLSerializable.kt:57]: There is a more elaborate function for this included at https://github.com/flytegg/twilight/blob/master/src/main/kotlin/gg/flyte/twilight/string/String.kt#L147

**Review by joshbker — CHANGES_REQUESTED** [review]
Please resolve the comments left

File: src/main/kotlin/gg/flyte/twilight/data/sql/SQLSerializable.kt (hunk 1 above)
Comment (kotlinsyntax) [reply src/main/kotlin/gg/flyte/twilight/data/sql/SQLSerializable.kt:57]: Ah Okay

**Comment by kotlinsyntax** [comment]
> Please resolve the comments left

resolved in [2db3f41](https://github.com/flytegg/twilight/pull/48/commits/2db3f416be3332fdb0bbad14591f269a1fe8e47c)

**Review by joshbker — APPROVED** [review]

**Pull request merged**